src/mono/mono/mini/method-to-ir.c

```diff
@@ -2307,6 +2303,20 @@ emit_not_supported_failure (MonoCompile *cfg)
 	mono_emit_jit_icall (cfg, mono_throw_not_supported, NULL);
 }
 
+static void
+emit_type_load_failure (MonoCompile* cfg, MonoClass* klass)
+{
+	char* class_name;
+	if (G_UNLIKELY (!klass)) 
+		class_name = mono_mem_manager_strdup (cfg->mem_manager, "Unknown class" );
+	else
+		class_name = mono_mem_manager_strdup (cfg->mem_manager, mono_class_full_name (klass));
+
+	MonoInst* iargs[1];
+	EMIT_NEW_LDSTRLITCONST (cfg, iargs [0], class_name);
```

Comment:
Is there a reason why you don't use `EMIT_NEW_CLASSCONST` instead? does it not work with failed types?